packages/beacon-node/test/unit/api/impl/validator/produceBlockV3.test.ts

```diff
@@ -41,23 +41,38 @@ describe("api/validator - produceBlockV3", function () {
     vi.clearAllMocks();
   });
 
-  const testCases: [routes.validator.BuilderSelection, number | null, number | null, number, string][] = [
-    [routes.validator.BuilderSelection.MaxProfit, 1, 0, 0, "builder"],
-    [routes.validator.BuilderSelection.MaxProfit, 1, 2, 1, "engine"],
-    [routes.validator.BuilderSelection.MaxProfit, null, 0, 0, "engine"],
-    [routes.validator.BuilderSelection.MaxProfit, 0, null, 1, "builder"],
-
-    [routes.validator.BuilderSelection.BuilderAlways, 1, 2, 0, "builder"],
-    [routes.validator.BuilderSelection.BuilderAlways, 1, 0, 1, "builder"],
-    [routes.validator.BuilderSelection.BuilderAlways, null, 0, 0, "engine"],
-    [routes.validator.BuilderSelection.BuilderAlways, 0, null, 1, "builder"],
-
-    [routes.validator.BuilderSelection.BuilderOnly, 0, 2, 0, "builder"],
-    [routes.validator.BuilderSelection.ExecutionOnly, 2, 0, 1, "execution"],
+  const testCases: [routes.validator.BuilderSelection, number | null, number | null, number, boolean, string][] = [
+    [routes.validator.BuilderSelection.MaxProfit, 1, 0, 0, false, "builder"],
+    [routes.validator.BuilderSelection.MaxProfit, 1, 2, 1, false, "engine"],
+    [routes.validator.BuilderSelection.MaxProfit, null, 0, 0, false, "engine"],
+    [routes.validator.BuilderSelection.MaxProfit, 0, null, 1, false, "builder"],
+    [routes.validator.BuilderSelection.MaxProfit, 0, null, 1, true, "builder"],
+    [routes.validator.BuilderSelection.MaxProfit, 1, 1, 1, true, "engine"],
+    [routes.validator.BuilderSelection.MaxProfit, 2, 1, 1, true, "engine"],
+
+    [routes.validator.BuilderSelection.BuilderAlways, 1, 2, 0, false, "builder"],
+    [routes.validator.BuilderSelection.BuilderAlways, 1, 0, 1, false, "builder"],
+    [routes.validator.BuilderSelection.BuilderAlways, null, 0, 0, false, "engine"],
+    [routes.validator.BuilderSelection.BuilderAlways, 0, null, 1, false, "builder"],
+    [routes.validator.BuilderSelection.BuilderAlways, 0, 1, 1, true, "engine"],
+    [routes.validator.BuilderSelection.BuilderAlways, 1, 1, 1, true, "engine"],
+    [routes.validator.BuilderSelection.BuilderAlways, 1, null, 1, true, "builder"],
+
+    [routes.validator.BuilderSelection.BuilderOnly, 0, 2, 0, false, "builder"],
+    [routes.validator.BuilderSelection.ExecutionOnly, 2, 0, 1, false, "engine"],
+    [routes.validator.BuilderSelection.BuilderOnly, 1, 1, 0, true, "builder"],
```

Comment:
this will be changed in followup PR where builderselection will be removed on the beacon side (didn't want to mix it in this PR), so just ignore for now, followup on builderselection coming shortly